DXRModules/DeusEx/Classes/DXRDatacubes.uc

```diff
@@ -752,11 +1766,15 @@ simulated function bool InfoDevsHasPass(#var(injectsprefix)InformationDevices id
         CriticalDelete(parser);
     }
 
-#ifdef injections
-    if( id.plaintext != "" ) {// TODO: wait maybe this works now with DXRInfoDevs?
-        ProcessStringHasPass(id.plaintext, hasPass, numHasPass);
+    #ifdef injections
+        injectID=id;
+    #else
+        injectID=#var(injectsprefix)InformationDevices(id);
```

Comment:
why don't we just change the argument type?

Reply:
This gets hit before datacubes are replaced in non-injections, but there might be freshly spawned PlainText Datacubes with passwords also spawned.  So there can be a mix of plain Datacubes and DXRInformationDevices